AppCenter/AppCenter/Internals/HttpClient/MSHttpClientProtocol.h

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <Foundation/Foundation.h>
+
+typedef void (^MSHttpRequestCompletionHandler)(NSHTTPURLResponse *response, NSData *data, NSError *error);
+
+@protocol MSHttpClientProtocol <NSObject>
+
+@required
+
+/**
+ * Send data to backend
+ *
+ * @param data A data instance that will be transformed request body.
+ * @param headers HTTP headers.
+ * @param url The endpoint to use in the HTTP request.
+ * @param method The HTTP method (verb) to use for the HTTP request (e.g. GET, POST, etc.).
+ * @param handler Completion handler.
+ */
+- (void)sendAsync:(nullable NSObject *)data
+              headers:(nullable NSDictionary<NSString *, NSString *> *)headers
+                  url:(NSURL *)url
+               method:(NSString *)method
```

Comment:
I wonder if it is a good idea to define a new type for HTTP method and use it instead of using `NSString` literal or variable (constant).

Reply:
Android uses String for method.

Reply:
I would prefer to use string because ultimately that is what it needs to be. But the caller can create const strings, which I would prefer

Reply:
Lets provide those constants though.

Reply:
It's quite predefined if that's HTTP verbs, constants make sense here to me.